### PR TITLE
Fix decals not affecting GI

### PR DESCRIPTION
**Pull request by EvgeniiG**
### Purpose of this PR
Decals were modifying the normal way too late. As a result, they were not affecting a bunch of stuff, including GI: https://fogbugz.unity3d.com/f/cases/1111046/

---
### Testing status

**Manual Tests**: Yes
**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/HDRP%252Fdecal-affect-gi/.yamato%252Fupm-ci-hdrp.yml%2523All_HDRP/531669/job

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: High (we have the same code copy-pasted 11 times. seriously?)

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
I have re-arrange some stuff as it was breaking specular occlusion in some case + causing compiler error due to uninitialize struct

File: com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitPass.template [hunk 1]
```diff
@@ -446,7 +446,7 @@ $include("SharedCode.template.hlsl")
         // compute world space normal
         GetNormalWS(fragInputs, normalTS, surfaceData.normalWS, doubleSidedConstants);
 
-        surfaceData.bentNormalWS = surfaceData.geomNormalWS; // better to use the geometric one for occlusion by default if no bentnormal map given
+        surfaceData.bentNormalWS = surfaceData.normalWS;
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitPass.template]: This contradicts the comments.

File: com.unity.render-pipelines.high-definition/Editor/Material/Eye/ShaderGraph/EyePass.template [hunk 2]
```diff
@@ -229,6 +229,8 @@ $include("SharedCode.template.hlsl")
          // compute world space normal
         GetNormalWS(fragInputs, irisNormalTS, surfaceData.irisNormalWS, doubleSidedConstants);
 
+        surfaceData.geomNormalWS = fragInputs.tangentToWorld[2];
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Editor/Material/Eye/ShaderGraph/EyePass.template]: I don't see why moving this up matters?

File: com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricPass.template [hunk 3]
```diff
@@ -236,6 +236,10 @@ $include("SharedCode.template.hlsl")
         // compute world space normal
         GetNormalWS(fragInputs, normalTS, surfaceData.normalWS, doubleSidedConstants);
 
+        surfaceData.geomNormalWS = fragInputs.tangentToWorld[2];
+        surfaceData.tangentWS = normalize(fragInputs.tangentToWorld[0].xyz);    // The tangent is not normalize in tangentToWorld for mikkt. TODO: Check if it expected that we normalize with Morten. Tag: SURFACE_GRADIENT
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricPass.template]: Moving this up is a mistake. Now decals do not affect surfaceData.tangentWS.

Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricPass.template]: Why move this up? This breaks the intuition that we apply decals after GetNormalWS.

Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricPass.template]: Additionally, surfaceData.tangentWS is only used for shading, and is not used for normal mapping. The shading code expects it to be a unit vector orthogonal to the normal (see orthonormalize call below).

File: com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxFData.hlsl [hunk 4]
```diff
@@ -141,6 +135,16 @@ void GetSurfaceAndBuiltinData(FragInputs input, float3 V, inout PositionInputs p
         sincos(surfaceData.anisotropyAngle, tangentTS.y, tangentTS.x);
         surfaceData.tangentWS = TransformTangentToWorld(tangentTS, input.tangentToWorld);
     }
+
+    #if HAVE_DECALS
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxFData.hlsl]: Why move this down?

**Pull request merged**